### PR TITLE
General: OpenPype modules publish plugins are registered in host

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Brief description
Module's publish plugin paths are registered during installation of OpenPype plugins.

## Description
Changed how module plugin paths are registered. Instead of using environment variables are plugin paths registered during installation of OpenPype plugins which is more suitable and possible dynamic.

## Additional info
This approach gives more dynamic approach of registering publish plugins (e.g. does not require restart of tray when module change state). Can be extended in future to install plugin paths per project and more dynamically, which is not as easy now because modules' plugin paths are there from moment when openpype starts.

PR is based on https://github.com/pypeclub/OpenPype/pull/3179 where creator plugins are installed same way. I wanted to separate the PRs as this change should be fully tested.

## Testing notes:
1. All modules should register their publish plugins in all hosts
2. The same must work for farm publishing, standalone publishing and webpublisher

### CODE REVIEW

File: openpype/pipeline/create/context.py [hunk 1]
```diff
@@ -861,6 +865,17 @@ def reset_plugins(self, discover_publish_plugins=True):
...
+
+            # Filter by host name
+            if (
+                creator_class.host_name
+                and creator_class.host_name != self.host_name
+            ):
+                self.log.info((
+                    "Creator's host name is not supported for current host {}"
+                ).format(creator_class.host_name, self.host_name))
+                continue
```
Comment (BigRoy) [inline openpype/pipeline/create/context.py:877]: For consistency, shouldn't we support similar host name filtering like Pyblish does? Like allowing `hosts = ["maya", "houdini"]` and maybe have default value of `hosts = ["*"]`?

Comment (iLLiCiTiT) [reply openpype/pipeline/create/context.py:877]: I don't think you can have same creator plugins for 2 different hosts....

Comment (BigRoy) [reply openpype/pipeline/create/context.py:877]: Ah, of course that might make a lot less sense.

**Review by kalisp — APPROVED** [review]
Tested in AE,PS, Nuke, Standalone publisher

**Pull request merged**